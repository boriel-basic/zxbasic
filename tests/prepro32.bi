DIM a As Ubyte = 1

#define MyMacro  \
    PRINT a; " MACRO START" \
    ASM \
    ld hl, _a \
    inc (hl) \
    END ASM \
    PRINT a; " MACRO END"

PRINT "HELLO WORLD"
MyMacro

