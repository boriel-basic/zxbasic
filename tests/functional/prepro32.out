#line 1 "prepro32.bi"
DIM a As Ubyte = 1



PRINT "HELLO WORLD"
 
    PRINT a; " MACRO START" 
    ASM 
    ld hl, _a 
    inc (hl) 
    END ASM 
    PRINT a; " MACRO END"

